.cmake/pyre_gsl.cmake

```
@@ -64,6 +64,8 @@ function(pyre_gslModule)
       target_include_directories(gslmodule PRIVATE ${MPI_CXX_INCLUDE_PATH})
       # and the mpi libraries
       target_link_libraries(gslmodule PRIVATE ${MPI_CXX_LIBRARIES})
+      # add the preprocessor macro
+      add_compile_definitions(WITH_MPI)
```

I think it's recommended to use target_compile_definitions for modern cmake
```suggestion
      target_compile_definitions(gslmodule PRIVATE WITH_MPI)
```

---

Thanks! It seems that I forked from an old version of pyre. I will redo it with a new pull request.